R/create_output_type_item.R

```diff
@@ -423,17 +483,29 @@ create_output_type_dist <- function(
     )
   }
 
-  check_prop_not_all_null(required, optional)
-  check_prop_type_const(required, optional)
-  check_prop_dups(required, optional)
+  if (pre_v4) {
+    check_prop_not_all_null(required, optional)
+    check_prop_type_const(required, optional)
+    check_prop_dups(required, optional)
```

Comment:
Do we not need these checks for the `required` field in v4?  I don't see them under the else condition below.

Reply:
I don't think so. The only thing that we should really check in the next condition is that `required` is not null.

Reply:
Yup not, needed because they were performing some dynamic checks on both `required` and `optional` properties which are now not required for output types